contents/tutorials/compare-aws-bedrock-prompts.md

```diff
@@ -0,0 +1,331 @@
+---
+title: How to compare AWS Bedrock prompts
+date: 2024-08-29
+author:
+  - lior-neu-ner
+tags:
+  - product analytics
+  - AI engineering
+---
+
+Evaluating LLM prompts is crucial for determining whether your modifications are genuinely enhancing your application. In this tutorial, we explore three methods to assess prompts by comparing their LLM outputs. Namely:
+
+1. [Quantitative metrics](#quantitative-metrics)
+2. [User feedback](#user-feedback)
+3. [Model-based evaluation](#model-based-evaluation)
+
+To show you how, we set up a basic Next.js app, implement the AWS Bedrock API, and capture events using PostHog.
+
+> While this tutorial focuses on [Next.js](/docs/libraries/next-js) and [Node](/docs/libraries/node), PostHog supports many different [SDKs](/docs/libraries) and [frameworks](/docs/frameworks). The concepts in this tutorial apply to all our supported SDKs and frameworks. 
+
+## Download the sample app
+
+We've created a sample app for this tutorial. You can download it from [Github](https://github.com/PostHog/aws-bedrock-compare-prompts-sample-app). 
+
+```bash
+git clone git@github.com:PostHog/aws-bedrock-compare-prompts-sample-app.git
+```
+
+To set your app up, first ensure [Node](https://nodejs.dev/en/learn/how-to-install-nodejs/) is install. Then run `npm install` to install all dependencies. 
+
+You must also ensure that you have properly configured your AWS credentials and region to use the AWS SDK for JavaScript. You can do this by calling `aws configure` using the [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).
+
+Once done, update the `BedrockRuntimeClient` initializer in `src/app/api/generate-llm-output/route.js` to use your preferred AWS region:
+
+```js file=src/app/api/generate-llm-output/route.js
+import {
+  BedrockRuntimeClient,
+  InvokeModelCommand,
+} from "@aws-sdk/client-bedrock-runtime";
+
+const client = new BedrockRuntimeClient({ region: "<YOUR_AWS_REGION>" }); // e.g. us-west-2
+
+// rest of the code
+```
+
+You'll also notice that we're using Meta's Llama 3 8B Instruct model. Make sure you have access to this model, or [request access](https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html) if you don't (you may need to change regions in AWS if it's not available. Alternatively, you can use a different Llama model). 
```

Comment:
It's fine to have this outside brackets, especially because it is multiple sentences.
```suggestion
You'll also notice that we're using Meta's Llama 3 8B Instruct model. Make sure you have access to this model, or [request access](https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html) if you don't. You may need to change regions in AWS if it's not available. Alternatively, you can use a different Llama model. 
```